rllib/agents/es/es_torch_policy.py

```diff
@@ -0,0 +1,108 @@
+# Code in this file is adapted from:
+# https://github.com/openai/evolution-strategies-starter.
+
+import gym
+import numpy as np
+
+import ray
+import ray.experimental.tf_utils
+from ray.rllib.evaluation.sampler import _unbatch_tuple_actions
+from ray.rllib.models import ModelCatalog
+from ray.rllib.policy.sample_batch import SampleBatch
+from ray.rllib.policy.torch_policy_template import build_torch_policy
+from ray.rllib.utils.filter import get_filter
+from ray.rllib.utils.framework import try_import_torch
+from ray.rllib.utils.torch_ops import convert_to_torch_tensor
+
+torch, _ = try_import_torch()
+
+
+def before_init(policy, observation_space, action_space, config):
+    policy.action_noise_std = config["action_noise_std"]
+    policy.preprocessor = ModelCatalog.get_preprocessor_for_space(
+        observation_space)
+    policy.observation_filter = get_filter(config["observation_filter"],
+                                           policy.preprocessor.shape)
+    policy.single_threaded = config.get("single_threaded", False)
+
+    def _set_flat_weights(policy, theta):
+        pos = 0
+        theta_dict = policy.model.state_dict()
+        new_theta_dict = {}
+
+        for k in sorted(theta_dict.keys()):
+            shape = policy.param_shapes[k]
+            num_params = int(np.prod(shape))
+            new_theta_dict[k] = torch.from_numpy(
+                np.reshape(theta[pos:pos + num_params], shape))
+            pos += num_params
+        policy.model.load_state_dict(new_theta_dict)
+
+    def _get_flat_weights(policy):
```

Comment:
Should these be common methods of TorchPolicy?

Reply:
Not sure, but it's high on my TODO list to unify these with tf/eager policies: Return a unified dict with identical keys/equally sorted list for two Policies that have equivalent Models.